packages/vite/src/node/optimizer/index.ts

```diff
@@ -1306,28 +1295,17 @@ export async function optimizedDepNeedsInterop(
   return depInfo?.needsInterop
 }
 
-const MAX_TEMP_DIR_AGE_MS = 24 * 60 * 60 * 1000
-export async function cleanupDepsCacheStaleDirs(
-  config: ResolvedConfig,
-): Promise<void> {
-  try {
-    const cacheDir = path.resolve(config.cacheDir)
-    if (fs.existsSync(cacheDir)) {
-      const dirents = await fsp.readdir(cacheDir, { withFileTypes: true })
-      for (const dirent of dirents) {
-        if (dirent.isDirectory() && dirent.name.includes('_temp_')) {
-          const tempDirPath = path.resolve(config.cacheDir, dirent.name)
-          const stats = await fsp.stat(tempDirPath).catch((_) => null)
-          if (
-            stats?.mtime &&
-            Date.now() - stats.mtime.getTime() > MAX_TEMP_DIR_AGE_MS
-          ) {
-            await removeDir(tempDirPath)
-          }
-        }
+async function cleanupDepsCacheStaleFiles(
+  oldFilesNames: string[],
+  depsCacheDir: string,
+  metadata: DepOptimizationMetadata,
+) {
+  for (const fileName of oldFilesNames) {
+    if (fileName !== 'package.json' && fileName !== '_metadata.json') {
+      const filePath = path.join(depsCacheDir, fileName)
+      if (!optimizedDepInfoFromFile(metadata, normalizePath(filePath))) {
+        fsp.unlink(filePath)
```

Comment:
With my suggested method, we don't need to rely on `optimizedDepInfoFromFile` and `normalizePath` too, since we're cleaning all files that aren't new and it implies being unused.

Reply:
Ah, I see, ok, let me try to simplify this

Reply:
Let me know if you had something like the latest in mind

Reply:
Ah I just read it and I understand how it works before now 😅 I didn't quite link that `outputFiles` is the same as `optimizedDepInfoFromFile`. I would be fine with the before code too, but let's go with the new changes then since you've made it. My bad.

Reply:
I think the new code is more clear though, so thanks for pushing me to do it 😅